packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -476,7 +476,7 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate,
     y += this.virtualizer.visibleRect.y;
 
     let key = this.virtualizer.keyAtPoint(new Point(x, y));
-    if (key == null) {
+    if (key == null || this.collection.size === 0) {
```

Comment:
When a list is empty, the key returned when dropping on it is ["placeholder"](https://github.com/adobe/react-spectrum/blob/7f69e65350a03023b2722b5cc293ee765c145ae4/packages/%40react-stately/layout/src/ListLayout.ts#L174). Check the collection size instead since that key name may be used by the user